dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityLiveMode.java

```diff
@@ -126,6 +129,13 @@ public final void serve(final OutputStream out) throws DotDataException, IOExcep
                 }
             }
 
+            //Validate if template is publish, if not remove page from cache
+            if (!UtilMethods.isSet(APILocator.getTemplateAPI().findLiveTemplate(htmlPage.getTemplateId(),APILocator.systemUser(),false).getInode())) {
```

Comment:
All our front end (delivery tier) calls need to use cache.  Please insure that this call is cached and also that the templateAPI has a 404 cache.  If it doesn't, you might want to just call the `VersionableAPI.findLiveVersion` which does have a 404 cache.

Reply:
Looking at the TemplateAPI and internally calls the versionableFactory that contains 404 cache as you mentioned.